build-setup/macos.md

```diff
@@ -19,12 +19,25 @@ umask 0002
 export JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_121.jdk/Contents/Home
 export PYTHONHOME=/Library/Frameworks/Python.framework/Versions/3.7
 export PATH=$JAVA_HOME/bin:$PYTHONHOME/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
-# Only required if building the C++ code directly using make - adjust depending on the location of your Git clone
+# Only required if building the C++ code directly using cmake - adjust depending on the location of your Git clone
 export CPP_SRC_HOME=$HOME/ml-cpp
 ```
 
 Note, that bash doesn't read `~/.bashrc` for login shells (which is what you get when you open a new Terminal window) - it reads `~/.bash_profile` or `~/.profile` so you should probably also create a `.bash_profile` file in your home directory which sources `.bashrc`.
 
+The build system uses CMake. To build (once the following dependencies have been installed) either call `cmake` directly from the top level of the source tree, e.g.
+```
+cmake -B cmake_build -DCMAKE_TOOLCHAIN_FILE=cmake/darwin-aarch64.cmake
+cmake --build cmake_build -v -j`nproc`
+```
+
+or, more simply, use Gradle
+```
+./gradlew :compile
+```
+
+Note that we configure the build to be of type `Release`, and specify the compiler flag `-g` in the cmake configuration files in order to obtain a fully optimized build along with debugging symbols. This is used in preference to `RelWithDebInfo` as `Release` generally gives a higher optimization level than `RelWithDebInfo` (`O3` vs `O2` respectively)
```

Comment:
If this is moved to in https://github.com/elastic/ml-cpp/blob/main/CONTRIBUTING.md#working-with-the-ml-cpp-codebase then there's no need to duplicate it for the different platforms.  Also it makes more sense in a "Working with the codebase" section because setting up a build machine is a one-off task per machine but people might want to refer back to how to build more frequently.